polaris_wyre/wyre/integration.py

```diff
@@ -0,0 +1,147 @@
+        in future ``GET /transaction`` responses.
+        **Polaris assumes ``Transaction.save()`` is called within this method.**
+        The memo saved to the transaction object _must_ be unique to the
+        transaction, since the anchor is expected to match the database record
+        represented by `transaction` with the on-chain transaction submitted.
+        This function differs from ``get_distribution_account()`` by allowing the
+        anchor to return any Stellar account that can be used to receive a payment.
+        This is ideal when the account used is determined by a custody service
+        provider that does not guarantee that the account provided will be the
+        account provided for future transactions.
+        :param request: the request that initiated the call to this function
+        :param transaction: the transaction a Stellar account and memo must be
+            saved to
+        """
+        stellar_account_address, user_id = self.wyre.get_account()
+        transaction.receiving_anchor_account = stellar_account_address
+        # TODO: Find a way to let the memo unique when integrating with Wyre
+        transaction.memo_type = Transaction.MEMO_TYPES.text
+        transaction.memo = user_id
+        transaction.save()
+
+    def submit_deposit_transaction(
+        self, transaction: Transaction, has_trustline: bool = True
+    ) -> dict:
+        """
+        Submit the transaction to the Stellar network using the anchor's custody
+        service provider and return the JSON body of the associated
+        ``GET /transaction/:id`` Horizon response.
+        If ``self.claimable_balances_supported`` is ``True``, Polaris may call
+        this method when the destination account does not yet have a trustline
+        to ``Transaction.asset``. In this case, the anchor should send the
+        deposit as a claimable balance instead of a payment or path payment. Use
+        the `has_trustline` parameter to determine which operations to use.
+        If ``self.claimable_balances_supported`` is ``False``, this method will only
+        be called when the destination account exists and has a trustline to
+        ``Transaction.asset``.
+        :param transaction: the ``Transaction`` object representing the Stellar
+            transaction that should be submitted to the network
+        :param has_trustline: whether or not the destination account has a trustline
+            for the requested asset
+        """
+        amount = round(
+            Decimal(transaction.amount_in) - Decimal(transaction.amount_fee),
+            transaction.asset.significant_decimals,
+        )
+        transfer_data = TransferData(
+            currency=transaction.asset.code,
+            amount=amount,
+            destination=f"stellar:{transaction.to_address}",
+        )
+
+        response_data = self.wyre.create_transfer(transfer_data)
+        transaction_id = self.wyre.get_stellar_transaction_id(response_data["id"])
```

Comment:
Wouldn't it make more sense to leave all knowledge of response data structure to the `Wyre` class like we discussed before?

Reply:
That makes sense. I'll return only the transaction id on `create_transfer` method and then pass it as a parameter to `get_stellar_transaction_id`. What do you think?